### PR TITLE
ceph: remove driveGroup support

### PR DESCRIPTION
**Pull request by BlaineEXE**
Remove features and design that supports adding OSDs to Ceph clusters
via `spec:driveGroups`. Update the Ceph upgrade doc that informs users
who currently use Drive Groups (we believe there are none of these
users) how to migrate to using the `spec:storage` config.

Resolves https://github.com/rook/rook/issues/7275

Revert "ceph: fix drive group deployment failure"
This reverts commit 76f1d9944e3ad5ecfb23d348abade9ab16b85159.

Revert "ceph: osd: add drive groups spec to cluster CR"
This reverts commit 7117fc12b752868cc4a3660d404e3bee01b109d4.

Revert "design: ceph orchestrator module add/remove OSDs"
This reverts commit 178187d035c219f404379257bc7759683ac91e51.

Signed-off-by: Blaine Gardner <blaine.gardner@redhat.com>

**Checklist:**

- [ ] **Commit Message Formatting**: Commit titles and messages follow guidelines in the [developer guide](https://rook.io/docs/rook/master/development-flow.html#commit-structure).
- [ ] **Skip Tests for Docs**: Add the flag for skipping the build if this is only a documentation change. See [here](https://github.com/rook/rook/blob/master/INSTALL.md#skip-ci) for the flag.
- [ ] **Skip Unrelated Tests**: Add a flag to run tests for a specific storage provider. See [test options](https://github.com/rook/rook/blob/master/INSTALL.md#test-storage-provider).
- [ ] Reviewed the developer guide on [Submitting a Pull Request](https://rook.io/docs/rook/master/development-flow.html#submitting-a-pull-request)
- [ ] Documentation has been updated, if necessary.
- [ ] Unit tests have been added, if necessary.
- [ ] Integration tests have been added, if necessary.
- [ ] Pending release notes updated with breaking and/or notable changes, if necessary.
- [ ] Upgrade from previous release is tested and upgrade user guide is updated, if necessary.
- [ ] Code generation (`make codegen`) has been run to update object specifications, if necessary.

### CODE REVIEW

File: Documentation/ceph-upgrade.md [hunk 1]
```diff
@@ -471,4 +475,36 @@ k8s.gcr.io/sig-storage/csi-resizer:v1.0.0
...
+
+1. If any of your specified Drive Groups use `host_pattern: '*'`, set `spec:storage:useAllNodes: true`.
+   1. If a drive group that uses `host_pattern: '*'` also sets `data_devices:all: true`, set
+     `spec:storage:useAllDevices: true`, and no more config migration should be necessary.
+1. If no Drive Groups use `host_pattern: '*'`, there are two basic options:
+   1. Determine which nodes apply to the Drive Group, then add each nodes to the 
+      `spec:storage:nodes` list.
+      1. Determine which nodes are already hosting OSDs using the below one-liner to list the nodes.
+          ```console
+          kr get pod --selector 'app==rook-ceph-osd' --output custom-columns='NAME:.metadata.name,NODE:.spec.nodeName,LABELS:.metadata.labels' --no-headers| grep -v ceph.rook.io/pvc | awk '{print $2}' | uniq
```
[the remark this answers is not in the document]
Comment (travisn) [reply Documentation/ceph-upgrade.md:504]: Do we really need to provide a complex command like this? It seems sufficient to tell them to figure out which nodes their OSDs are running on, it's not a difficult k8s task.

**Review by travisn — CHANGES_REQUESTED** [review]

File: Documentation/ceph-upgrade.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (BlaineEXE) [reply Documentation/ceph-upgrade.md:504]: Sorry, `kr` is my `kubectl -n rook-ceph` alias. I'll fix that. I don't think we need the command, but I already wrote it out for people to use, so we may as well keep it?

File: pkg/operator/ceph/cluster/osd/osd.go [hunk 2]
```diff
@@ -468,6 +450,10 @@ func (c *Cluster) startNodeStorageProvisioners(config *provisionConfig) {
 			continue
 		}
 
+		// update the orchestration status of this node to the starting state
+		status := OrchestrationStatus{Status: OrchestrationStatusStarting}
```
[the remark this answers is not in the document]
Comment (BlaineEXE) [reply pkg/operator/ceph/cluster/osd/osd.go:454]: I think we still need it. There is an equivalent code block in the PVC provisioning method.

[the remark this answers is not in the document]
Comment (BlaineEXE) [reply pkg/operator/ceph/cluster/osd/osd.go:454]: PVC equivalent is here, FYI: https://github.com/rook/rook/blob/10547a4e2fc8fc44b5f7d791e53e212b62e886eb/pkg/operator/ceph/cluster/osd/osd.go#L352

[the remark this answers is not in the document]
Comment (travisn) [reply pkg/operator/ceph/cluster/osd/osd.go:454]: Ah I see now setting this status already existed, but the code just moved around

**Review by travisn — APPROVED** [review]
> Sorry, kr is my kubectl -n rook-ceph alias. I'll fix that. I don't think we need the command, but I already wrote it out for people to use, so we may as well keep it?

There is a balance between requiring some level of proficiency with K8s and complex commands in the docs. I tend to avoid complex commands, but it's more of a personal preference, I'm ok if you keep it.

**Review by leseb — APPROVED** [review]

**Pull request merged**